crates/shared/src/current_block.rs

```diff
@@ -138,6 +137,45 @@ where
             .context("failed to get current block number")?
             .as_u64())
     }
+
+    /// get blocks defined by the range (inclusive)
+    /// if successful, function guarantees full range of blocks in Result (does not return partial results)
+    async fn blocks(&self, range: RangeInclusive<u64>) -> Result<Vec<BlockNumberHash>> {
+        ensure!(
+            range.start() <= range.end(),
+            "invalid range: {} - {}",
+            range.start(),
+            range.end()
+        );
```

Comment:
ubernit (feel free to ignore): Should this maybe be a `debug_assert!` in that we shouldn't ever call it if that is the case?

Reply:
No strong opinion here tbh. I guess it's ensure because it is used in the code where weird ranges could be provided (for example if loadbalanced node returns weird `current_block`). In those cases I might prefer to skip the event update, because there is a high probability that _normal_ `current_block` would be received in next iteration of events update.